lib/workers/global/config/parse/file.spec.ts

```diff
@@ -113,22 +113,112 @@ describe('workers/global/config/parse/file', () => {
       expect(logger.fatal).toHaveBeenCalledWith(
         `Error parsing config file due to unresolved variable(s): CI_API_V4_URL is not defined`
       );
-      expect(mockProcessExit).toHaveBeenCalledWith(1);
+      expect(processExitSpy).toHaveBeenCalledWith(1);
     });
 
     it.each([
       ['invalid config file type', './file.txt'],
       ['missing config file type', './file'],
     ])('fatal error and exit if %s', async (fileType, filePath) => {
-      const mockProcessExit = jest
-        .spyOn(process, 'exit')
-        .mockImplementationOnce(() => undefined as never);
+      processExitSpy.mockImplementationOnce(() => undefined as never);
       const configFile = upath.resolve(tmp.path, filePath);
       fs.writeFileSync(configFile, `{"token": "abc"}`, { encoding: 'utf8' });
       await file.getConfig({ RENOVATE_CONFIG_FILE: configFile });
-      expect(mockProcessExit).toHaveBeenCalledWith(1);
+      expect(processExitSpy).toHaveBeenCalledWith(1);
       expect(logger.fatal).toHaveBeenCalledWith('Unsupported file type');
       fs.unlinkSync(configFile);
     });
+
+    it('removes the config file if RENOVATE_CONFIG_FILE & RENOVATE_X_DELETE_CONFIG_FILE are set', async () => {
+      fsRemoveSpy.mockImplementationOnce(() => {
+        // no-op
+      });
+      processExitSpy.mockImplementationOnce(() => undefined as never);
+      const configFile = upath.resolve(tmp.path, './config.json');
+      fs.writeFileSync(configFile, `{"token": "abc"}`, { encoding: 'utf8' });
+
+      await file.getConfig({
+        RENOVATE_CONFIG_FILE: configFile,
+        RENOVATE_X_DELETE_CONFIG_FILE: 'true',
+      });
+
+      expect(processExitSpy).not.toHaveBeenCalled();
+      expect(fsRemoveSpy).toHaveBeenCalledTimes(1);
+      expect(fsRemoveSpy).toHaveBeenCalledWith(configFile);
+      fs.unlinkSync(configFile);
+    });
+  });
+
+  describe('deleteConfigFile()', () => {
+    it.each([[undefined], [' ']])(
+      'skip when RENOVATE_CONFIG_FILE is not set ("%s")',
+      async (configFile) => {
+        await file.deleteConfigFile({ RENOVATE_CONFIG_FILE: configFile });
+
+        expect(fsRemoveSpy).toHaveBeenCalledTimes(0);
+      }
+    );
+
+    it('skip when config file does not exist', async () => {
+      fsPathExistsSpy.mockResolvedValueOnce(false as never);
```

Comment:
```suggestion
      fsPathExistsSpy.mockResolvedValueOnce(false);
```
do we really need that cast?

Reply:
ok, then something is wrong with the spy or typescript. 🤔

Reply:
can we resolve this or is there anything else we can do about it ?